contracts/asset-proxy/contracts/src/StaticCallProxy.sol

```diff
@@ -18,181 +18,57 @@
-                let callTarget := and(
-                    calldataload(add(assetDataOffset, 36)),
-                    0x000000000000000000000000ffffffffffffffffffffffffffffffffffffffff
-                )
-
-                // Perform `callTarget.staticcall(staticCallData)`
-                let success := staticcall(
-                    gas,                         // forward all gas
-                    callTarget,                  // call address `callTarget`
-                    0,                           // pointer to start of input
-                    staticCallDataLen,           // length of input
-                    0,                           // start of memory can be safely overwritten
-                    0                            // don't copy output to memory
-                )
-
-                // Copy entire output to start of memory
-                let outputLen := returndatasize()
-                returndatacopy(
-                    0,                // copy to memory at 0
-                    0,                // copy from return data at 0
-                    outputLen         // copy all return data
-                )
-
-                // Revert with reason given by `callTarget` if staticcall is unsuccessful
-                if iszero(success) {
-                    revert(0, outputLen)
-                }
-
-                // Calculate hash of output
-                let callResultHash := keccak256(0, outputLen)
-
-                // In order to find the offset to `expectedCallResultHash`, we must add:
-                // assetDataOffset
-                // + 32 (assetData len)
-                // + 4 (proxyId)
-                // + 32 (callTarget)
-                // + 32 (staticCallDataOffset)
-                let expectedResultHash := calldataload(add(assetDataOffset, 100))
-
-                if sub(callResultHash, expectedResultHash) {
-                    // Revert with `Error("UNEXPECTED_STATIC_CALL_RESULT")`
-                    mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                    mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                    mstore(64, 0x0000001d554e45585045435445445f5354415449435f43414c4c5f524553554c)
-                    mstore(96, 0x5400000000000000000000000000000000000000000000000000000000000000)
-                    revert(0, 100)
-                }
-
-                // Return if output matched expected output
-                return(0, 0)
+        // Decode params from `assetData`
+        (
+            address staticCallTarget,
+            bytes memory staticCallData,
+            bytes32 expectedReturnDataHash
+        ) = abi.decode(
+            assetData.sliceDestructive(4, assetData.length),
+            (address, bytes, bytes32)
+        );
+
+        // Execute staticcall
+        (bool success, bytes memory returnData) = staticCallTarget.staticcall(staticCallData);
```

Comment:
Should we be doing an `extcodesize()` check to safeguard people from accidentally calling EOAs? Could be a problem if people expect a `keccak(bytes(0))` result.

At the very least, we should add a test that confirms that calling an EOA results in `keccak(bytes(0))`.

Reply:
I will add a test for this, but it feels a bit excessive to me to add an `extcodesize` or `extcodehash` check here. You really have to know the interface of and trust the `staticCallTarget` beforehand in order to use this.